lib/platform.h

```diff
@@ -43,24 +43,24 @@ namespace cppcheck {
         static long long min_value(int bit) {
             if (bit >= 64)
                 return LLONG_MIN;
-            return -(1LL << (bit-1));
+            return -(1LL << (static_cast<long long>(bit - 1)));
```

Comment:
I really do not understand why this would be any better. The value of `bit` is typically 8/16/32

Reply:
Signature of method is an int (at least 16 bit) but is being converted in the method to a long long (at least 64bit). https://en.cppreference.com/w/cpp/language/types ('Properties' table)